packages/slate-react/src/utils/get-children-decorations.js

```diff
@@ -0,0 +1,129 @@
+import { Set } from 'immutable'
+
+/**
+ * Split the decorations in lists of relevant decorations for each child.
+ *
+ * @param {Node} node
+ * @param {List} decorations
+ * @return {Array<List<Decoration>>}
+ */
+
+function getChildrenDecorations(node, decorations) {
+  const activeDecorations = Set().asMutable()
+  const childrenDecorations = []
+
+  orderChildDecorations(node, decorations).forEach(item => {
+    if (item.isRangeStart) {
+      // Item is a decoration start
+      activeDecorations.add(item.decoration)
+    } else if (item.isRangeEnd) {
+      // item is a decoration end
+      activeDecorations.remove(item.decoration)
+    } else {
+      // Item is a child node
+      childrenDecorations.push(activeDecorations.toList())
```

Comment:
Why to list? or why were decorations a List?

Reply:
The List type originates from the implementation of `node.getDecorations` and `Range.createList`. Do you think changing it to a Set or Array is better?

Reply:
No, I made a mistake here.  I thought creating Set is faster than creating List. But it is wrong, creating List is indeed faster without checking same member.